iscsi/get_test.go

```diff
@@ -0,0 +1,157 @@
+			stat: &iscsi.Stats{
+				Name: "iqn.2003-01.org.linux-iscsi.osd1.x8664:sn.8888bbbbddd0",
+				Tpgt: []iscsi.TPGT{
+					{
+						Name:     "tpgt_1",
+						TpgtPath: "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2003-01.org.linux-iscsi.osd1.x8664:sn.8888bbbbddd0/tpgt_1",
+						IsEnable: true,
+						Luns: []iscsi.LUN{
+							{
+								Name:       "lun_0",
+								LunPath:    "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2003-01.org.linux-iscsi.osd1.x8664:sn.8888bbbbddd0/tpgt_1/lun/lun_0",
+								Backstore:  "rd_mcp",
+								ObjectName: "ramdisk_lio_1G",
+								TypeNumber: "0",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			stat: &iscsi.Stats{
+				Name: "iqn.2003-01.org.linux-iscsi.osd1.x8664:sn.abcd1abcd2ab",
+				Tpgt: []iscsi.TPGT{
+					{
+						Name:     "tpgt_1",
+						TpgtPath: "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2003-01.org.linux-iscsi.osd1.x8664:sn.abcd1abcd2ab/tpgt_1",
+						IsEnable: true,
+						Luns: []iscsi.LUN{
+							{
+								Name:       "lun_0",
+								LunPath:    "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2003-01.org.linux-iscsi.osd1.x8664:sn.abcd1abcd2ab/tpgt_1/lun/lun_0",
+								Backstore:  "iblock",
+								ObjectName: "block_lio_rbd1",
+								TypeNumber: "0",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			stat: &iscsi.Stats{
+				Name: "iqn.2016-11.org.linux-iscsi.igw.x86:dev.rbd0",
+				Tpgt: []iscsi.TPGT{
+					{
+						Name:     "tpgt_1",
+						TpgtPath: "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2016-11.org.linux-iscsi.igw.x86:dev.rbd0/tpgt_1",
+						IsEnable: true,
+						Luns: []iscsi.LUN{
+							{
+								Name:       "lun_0",
+								LunPath:    "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2016-11.org.linux-iscsi.igw.x86:dev.rbd0/tpgt_1/lun/lun_0",
+								Backstore:  "fileio",
+								ObjectName: "file_lio_1G",
+								TypeNumber: "1",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			stat: &iscsi.Stats{
+				Name: "iqn.2016-11.org.linux-iscsi.igw.x86:sn.ramdemo",
+				Tpgt: []iscsi.TPGT{
+					{
+						Name:     "tpgt_1",
+						TpgtPath: "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2016-11.org.linux-iscsi.igw.x86:sn.ramdemo/tpgt_1",
+						IsEnable: true,
+						Luns: []iscsi.LUN{
+							{
+								Name:       "lun_0",
+								LunPath:    "../sysfs/fixtures/kernel/config/target/iscsi/iqn.2016-11.org.linux-iscsi.igw.x86:sn.ramdemo/tpgt_1/lun/lun_0",
+								Backstore:  "rbd",
+								ObjectName: "iscsi-images-demo",
+								TypeNumber: "0",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	readTests := []struct {
+		read  uint64
+		write uint64
+		iops  uint64
+	}{
+		{10325, 40325, 204950},
+		{20095, 71235, 104950},
+		{10195, 30195, 301950},
+		{1504, 4733, 1234},
+	}
+
+	sysfs, err := iscsi.NewFS("../fixtures")
+	if err != nil {
+		t.Fatalf("failed to access xfs fs: %v", err)
+	}
+	sysfsStat, err := sysfs.ISCSIStats()
+	if err != nil {
+		t.Errorf("unexpected test fixtures")
+	}
+
```

Comment:
Should add a check here to make sure that the returned stats instance is the expected length.  It looks like the fixtures were accidentally removed in one of the recent commits, but the test still passes because a zero length stats object just skips the detailed checks.

Reply:
I need to rewrite this and add the fixture back